File: MyMindNotes/app/src/main/java/com/android/mymindnotes/model/retrofit/DeleteDiaryApi.java
package com.android.mymindnotes.model.retrofit;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.Path;

public interface DeleteDiaryApi {
    @DELETE("/api/diary/delete/{diary_number}")
    Call<Map<String, Object>> deleteDiary(@Path("diary_number") int diary_number);
}

File: MyMindNotes/data/src/main/java/com/android/mymindnotes/data/retrofit/api/diary/DeleteDiaryApi.kt
package com.android.mymindnotes.data.retrofit.api.diary

import com.android.mymindnotes.core.dto.DeleteDiaryResponse
import retrofit2.http.DELETE
import retrofit2.http.Path

interface DeleteDiaryApi {
    @DELETE("/api/diary/delete/{diary_number}")
    suspend fun deleteDiary(
        @Path("diary_number") diary_number: Int
    ): DeleteDiaryResponse
}